topic-operator/src/main/java/io/strimzi/operator/topic/v2/ReplicasChangeClient.java

```diff
@@ -0,0 +1,373 @@
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.net.URI;
+import java.net.URLEncoder;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpResponse;
+import java.security.KeyStore;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.time.Duration;
+import java.time.temporal.ChronoUnit;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+import static io.strimzi.api.kafka.model.topic.ReplicasChangeState.ONGOING;
+import static io.strimzi.api.kafka.model.topic.ReplicasChangeState.PENDING;
+import static io.strimzi.operator.topic.v2.TopicOperatorUtil.buildBasicAuthValue;
+import static io.strimzi.operator.topic.v2.TopicOperatorUtil.getFileContent;
+import static io.strimzi.operator.topic.v2.TopicOperatorUtil.hasReplicasChange;
+import static io.strimzi.operator.topic.v2.TopicOperatorUtil.topicNames;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.stream.Collectors.groupingBy;
+
+/**
+ * Replicas change client for Cruise Control API.
+ * <br/><br/>
+ * The REST endpoints are {@code topic_configuration} to request replication factor changes and {@code user_tasks} 
+ * to check the asynchronous execution result. Cruise Control runs one task execution at a time, additional requests 
+ * are queued up to {@code max.active.user.tasks}.
+ * <br/><br/>
+ * At any given time, a KafkaTopic.spec.replicas change can be in one of the following states:
+ * <br/>
+ * <ul><li>Pending: Not in Cruise Control's task queue (not yet sent or request error).</li>
+ * <li>Ongoing: In Cruise Control's task queue, but execution not started, or not completed.</li>
+ * <li>Completed: Cruise Control's task execution completed (target replication factor reconciled).</li></ul>
+ */
+public class ReplicasChangeClient {
+    private static final ReconciliationLogger LOGGER = ReconciliationLogger.create(ReplicasChangeClient.class);
+    private static final String USER_TASK_ID_HEADER = "User-Task-ID";
+    private static final long REQUEST_TIMEOUT_SEC = 60;
+    
+    private final TopicOperatorConfig config;
+    private final ExecutorService httpClientExecutor;
+    private final ObjectMapper mapper;
+    
+    /**
+     * Create a new replicas change client instance.
+     * 
+     * @param config Topic Operator configuration.
+     */
+    public ReplicasChangeClient(TopicOperatorConfig config) {
+        this.config = config;
+        this.httpClientExecutor = Executors.newCachedThreadPool();
+        this.mapper = new ObjectMapper();
+    }
+
+    /**
+     * Send a topic_configuration request to create a task for replication factor change of one or more topics.
```

Comment:
It does more than this though: It also updated the statuses. That needs to be mentioned. And it makes this more than a strict "client", because it builds in some of the logic for the status.

Reply:
Right, let me add this important note to both requestPendingChanges and requestOngoingChanges. I'm also renaming client to handler, as I think it better reflects its implementation.